app.rb

```diff
@@ -0,0 +1,135 @@
+require_relative './book'
+require_relative './teacher'
+require_relative './student'
+require_relative './rental'
+
+class App
+  def initialize
+    @persons = []
+    @books = []
+    @rentals = []
+  end
+
+  # 1 - list all books
+  def books_list
+    if @books.length.positive?
+      @books.each do |book|
+        puts " Title: #{book.title}, Author: #{book.author}"
+      end
+    else
+      puts 'No books added yet!'
+    end
+  end
+
+  # 2 - List all people'
+  def person_list
+    @persons.each do |individual|
+      # binding.pry
+      puts "[#{individual.class}] Name: #{individual.name}, Age: #{individual.age}"
+    end
+  end
+
+  # '3 - Create a teacher',
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @persons << Teacher.new(specialization, name, age)
+    puts 'create teacher'
+  end
+
+  # '3 - Create a student',
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp != 'n'
+    @persons << Student.new(name, age, parent_permission)
+    puts 'create student'
+  end
```

Comment:
- [ ] great job cleaning up the main.rb file 👍 in this method text is printed, information is gathered and processed, this is not according to the single responsibility principle. try to split this into smaller easier to maintain methods. have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.